transactions/WFLY-14869-microprofile-lra.adoc

```diff
@@ -0,0 +1,123 @@
+= Support for MicroProfile LRA
+:author:            Martin Stefanko
+:email:             mstefank@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+:keywords:          transaction, saga, lra, microprofile
+
+== Overview
+
+The goal of this RFE is providing the Galleon feature pack for WildFly which integrates the
+https://github.com/jbosstm/narayana/tree/master/rts/lra[Narayana implementation] of the
+https://github.com/eclipse/microprofile-lra[MicroProfile LRA specification].
+
+The MicroProfile LRA is a transaction processing pattern well suited for microservice development based on the saga
+pattern. The motivation of such processing is summarized in the
+https://github.com/eclipse/microprofile-lra/blob/1.0/README.adoc#motivation[MicroProfile Spec readme].
+
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-14869
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-1496
+
+=== Dev Contacts
+
+* mailto:mstefank@redhat.com[{author}]
+
+=== QE Contacts
+
+* mailto:fburzigo@redhat.com[Fabio Burzigotti]
+
+=== Testing By
+
+* [x] Engineering
+* [ ] QE
+
+
+=== Affected Projects or Components
+
+* Transactions
+* WildFly
+
+=== Other Interested Projects
+
+* MicroProfile
+* Galleon
+* WildFly Operator
+
+=== Relevant Installation Types
+
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+* [ ] Managed domain
+* [ ] OpenShift s2i
+* [x] Bootable jar
+
+== Requirements
+
+=== Hard Requirements
+
+The feature pack will integrate the LRA specification as the Galleon feature pack.
+The Galleon feature pack will be available for provisioning.
+The provisioned feature pack will provide all functionality defined in LRA specification for users in WildFly.
+
+The feature pack will be split to two Galleon layers (i.e., two separate subsystems - microprofile-lra-coordinator and microprofile-lra-participant).
```

Comment:
Is not wanting both a common use case? Is there a big downside to including both when you only need one?

IOW should we have just one layer? Should we have three, with the 3rd being 'microprofile-lra' that depends on the other two?

Having one still allows us to respond to user demand by moving to 3 in the future. Starting with 2 also lets us respond to user demand by moving to 3 in the future, but it also commits us to having 3.

Reply:
Most common use case would be only the participant. Coordinator is a REST endpoint that can be started separately. Basically similar to message broker. About putting them together I don't know. I will need to see how users will use them. We can always introduce thrid `microprofile-lra` if needed later.